instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/internal/JdbcConnectionUrlParser.java

```diff
@@ -789,6 +789,35 @@ DbInfo.Builder doParse(String jdbcUrl, DbInfo.Builder builder) {
       }
       return builder.name(instance);
     }
+  },
+
+  DATADIRECT("datadirect", "tibcosoftware") {
+    @Override
+    DbInfo.Builder doParse(String jdbcUrl, DbInfo.Builder builder) {
+      int typeEndIndex = jdbcUrl.indexOf(':');
+      int subtypeEndIndex = jdbcUrl.indexOf(':', typeEndIndex + 1);
+
+      if (subtypeEndIndex == -1) {
+        return builder;
+      }
+
+      String subtype = jdbcUrl.substring(typeEndIndex + 1, subtypeEndIndex);
+      builder.subtype(subtype);
+
+      if (subtype.equals("sqlserver")) {
```

Comment:
are these all possible subtypes? or could you have something like `jdbc:datadirect:mariadb://server_name:3306`?

Reply:
I believe all of the subtypes are listed here: https://docs.progress.com/bundle/datadirect-connect-jdbc-51/page/URL-Formats-DataDirect-Connect-for-JDBC-Drivers.html (I didn't include support for sybase or informix in the PR).

(my understanding is that these are real drivers as opposed to wrappers)